rules/rbe_repo.bzl

```diff
@@ -0,0 +1,597 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Repository Rules to generate toolchain configs for an RBE-ubuntu16-04 container.
+
+Exposes the rbe_autoconfig macro that does the following:
+- Pulls an rbe-ubuntu 16_04 image
+- Loads the image in the local registry
+- Starts up a container using the rbe-ubuntu 16_04 image mounting the current project
+- Installs the current version of Bazel (one currently running) on the container
+  (or the one passed as arg).
+- Runs a bazel command to build the local_config_cc
+  remote repository, inside the container.
+- Extracts local_config_cc produced files (inside the container) to the produced
+  remote repository.
+- Optionally copies the local_config_cc produced files to the project srcs under the
+  given output_base directory
+
+Add to your WORKSPACE file the following:
+
+  load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive", "http_file")
+
+  http_archive(
+    name = "bazel_toolchains",
+    urls = [
+      "https://mirror.bazel.build/github.com/bazelbuild/bazel-toolchains/archive/<latest_release>.tar.gz",
+      "https://github.com/bazelbuild/bazel-toolchains/archive/<latest_release>.tar.gz",
+    ],
+    strip_prefix = "bazel-toolchains-<latest_commit>",
+    sha256 = "<sha256>",
+  )
+
+  load(
+    "@bazel_toolchains//repositories:repositories.bzl",
+    bazel_toolchains_repositories = "repositories",
+  )
+
+  bazel_toolchains_repositories()
+
+  load("@bazel_toolchains//rules:rbe_repo.bzl", "rbe_autoconfig")
+
+  rbe_autoconfig(
+    name = "rbe_default",
+    # Optional. See below.
+    output_base = "rbe-configs"
+  )
+
+For values of <latest_release> and other placeholders above, please see
+the WORKSPACE file in this repo.
+
+This rule depends on the value of the environment variable "RBE_AUTOCONF_ROOT"
+when output_base is used.
+This env var should be set to point to the absolute path root of your project.
+Use the full absolute path to the project root (i.e., no '~', '../', or
+other special chars).
+
+There are two modes of using this repo rules:
+  1 - When output_base set (recommended), running the repo rule target will copy
+    (env var "RBE_AUTOCONF_ROOT" is required) the toolchain config files to the
```

Comment:
Why not take RBE_AUTOCONF_ROOT as another attribute in this repository rule instead of an environment variable?

Reply:
Because people would then have to submit their code to their repo with values that are machine specific.